rendering_resource_manager_service/session/management/job_manager.py

```diff
@@ -145,21 +146,24 @@ def allocate(self, session, job_information):
                 session.job_id = re.findall('\\d+', error)[0]
                 log.info(1, 'Allocated job ' + str(session.job_id))
             else:
+                session.status = SESSION_STATUS_FAILED
+                session.save()
                 log.error(error)
                 response = json.dumps({'contents': error})
                 return [400, response]
             process.stdin.close()
             session.status = SESSION_STATUS_SCHEDULED
             session.save()
             response = json.dumps({'message': 'Job scheduled', 'jobId': session.job_id})
-            return [200, response]
+            status = [200, response]
```

Comment:
Because the code was not executing the 'finally' and the lock was not released. That said, the return is a bug. Now fixed.